boards/nucleo-l476/include/periph_conf.h

```diff
@@ -37,6 +39,12 @@ extern "C" {
 /* 0: no external low speed crystal available,
  * 1: external crystal available (always 32.768kHz) */
 #define CLOCK_LSE           (1)
+/* 0: enable MSI only if HSE isn't available
```

Comment:
Maybe these new defines could also be added to other L4 based boards ? (nucleo32-l432, b-l475e-iotnode)

Reply:
I see that LSE is enables for the boards but as we know some boards don't have an external clock source (crystal etc.) So I'd say to better enable LSI as a general rule.

Reply:
>... you changed the board config to only one, is this intended ?

more or less - `b-l475e-iot01` wasn't there when I opened this PR. And I haven't checked (yet) if all three of the L4s support the MSI-LSE sync feature in the same way.
>Maybe these new defines could also be added to other L4 based boards ? 

Probably yes, I will check the Datasheets now and update the other configs if appropriate.

>...as we know some boards don't have an external clock source (crystal etc.) So I'd say to better enable LSI as a general rule.

Generally agree, but this file belongs to the board (nucleo-l476 in this case) and thus it should be clear if it has an external crystal. If another board comes without LSE the [*lfclk functions of stmclk_common.c](https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32_common/stmclk_common.c#L74) should take care of prefering LSI over LSE.

Reply:
> Generally agree, but this file belongs to the board (nucleo-l476 in this case) and thus it should be clear if it has an external crystal. If another board comes without LSE the *lfclk functions of stmclk_common.c should take care of prefering LSI over LSE.

The function you mention relies on the define just above the line we're commenting. The problem is actually that some boards have LSE, some not, it depends on the revision and I think it doesn't worth to create another define to know which revision is the board in question.

Reply:
So you say that some boards that currently define CLOCK_LSE 1 don't actually have an external crystal? This would be bad indeed. But then it's more of a general problem with the RIOT configuration and not directly related to this PR because I didn't actually change this compared to master.
>...some boards have LSE, some not, it depends on the revision

Do you know if this is also the case for one of the currently supported L4s? Is this reflected by their datasheets (e.g. multiple revisions that state differing configuation related to LSE)?

Reply:
> So you say that some boards that currently define CLOCK_LSE 1 don't actually have an external crystal? This would be bad indeed. But then it's more of a general problem with the RIOT configuration and not directly related to this PR because I didn't actually change this compared to master.

Yes, it's a general problem, but as you're changing the interface to generalise it for the current configuration, it's very possible that boards without crystal will just fail.

> Do you know if this is also the case for one of the currently supported L4s? Is this reflected by their datasheets (e.g. multiple revisions that state differing configuation related to LSE)?

Yes, the [datasheet](http://www.st.com/content/ccc/resource/technical/document/user_manual/98/2e/fa/4b/e0/82/43/b7/DM00105823.pdf/files/DM00105823.pdf/jcr:content/translations/en.DM00105823.pdf) (section 6.7) state what is the configuration according to the revision.

Reply:
>...it's very possible that boards without crystal will just fail

like they would also fail with current master ;) 
Nonetheless thanks for pointing out that issue here. I disabled LSE by default on l476 now. The other boards should be safe with LSE enabled by default - see comment below. Do you possess one of the l476 with HW-disabled LSE or do I need to whip out my soldering iron for testing?